src/client.c

```diff
@@ -64,11 +64,11 @@ void client_send(char *data) {
     int length = strlen(data);
     int header_size = htonl(length);
     if (client_sendall(sd, (char*)&header_size, sizeof(header_size)) == -1) {
-        perror("client_sendall");
+        SHOWERROR("client_sendall");
         exit(1);
     }
     if (client_sendall(sd, data, length) == -1) {
-        perror("client_sendall");
+		SHOWERROR("client_sendall");
```

Comment:
Please, change the tabs to spaces. The code will be hard to read with mixed indentation levels.